frontend/src/stimulus/controllers/hover-card-trigger.controller.ts

```diff
@@ -60,9 +60,25 @@ export default class HoverCardTriggerController extends ApplicationController {
   private cardMouseLeaveBound= this.onMouseLeave.bind(this);
   private cardMouseEnterBound= this.onMouseEnter.bind(this);
 
+  /**
+   * Attempting to show hover cards on touch-based devices often leads to quirky and unwanted
+   * behavior. As all information in our hover cards is optional, we'd rather only show them
+   * on devices that primarily support a mouse cursor. That means that hovering is a reliable
+   * option for interaction.
+   */
+  shouldRegisterHoverEventListeners() {
+    const browserSupportsHovering = window.matchMedia('(hover: hover)').matches;
+    // The browser on the CI claims not to support hovering, so we must detect whether we are in a test environment:
+    const isTestEnv = document.body.classList.contains('env-test');
```

Comment:
I am not really happy about this because of two reasons:

1. Adding extra logic for a test case feels like a smell. 
2. If the class name changes, this will break.

I don't have an idea what to do about the first point. We probably have to live with it. 😕
 For the second point, we could at least use one of the [helpers](https://github.com/opf/openproject/blob/release/15.3/frontend/src/app/core/setup/globals/openproject.ts#L66) that exists and call `window.OpenProject.environment === 'test'`

Reply:
Yes, it does feel like a smell. Previously, I patched `window.matchMedia` in a `before` block of the spec, but I could not get this executed before the page loads. Since the listeners are added whenever the DOM element appears, overriding `matchMedia` must be done as quickly as possible.

On top of that, having a messy `page.execute_script` block in a test case is a smell, too. When the code changes, the monkey patch must be adapted, too. So in the end, this seemed like the cleaner alternative for this case 🤔 But I am very open to suggestions if we can come up with an alternate approach.

As for 2: neat, I didn't know about this helper. Will use that!